### ISSUE TITLE
how to provide RouterContext for ios side correctly 0.5.0 in update

### ISSUE THREAD

**Issue by adelsaramii** [xxfast/Decompose-Router#70]
i saw your example about platform-configurations Ios but there is no sample and im lost
i give you better look from my code and please guide me how to implement ios part

Class main.ios.kt

    fun MainViewController(routerContext : RouterContext) = ComposeUIViewController {

        CompositionLocalProvider(LocalRouterContext provides routerContext) {

            App()

        }

    }

swift file ContentView.swift

    import UIKit
    import SwiftUI
    import shared

    struct ComposeView: UIViewControllerRepresentable {
       func makeUIViewController(context: Context) -> UIViewController {
           Main_iosKt.MainViewController()
       }

       func updateUIViewController(_ uiViewController: UIViewController, context: Context) {}
    }

    struct ContentView: View {
       var body: some View {
           ComposeView()
                   .ignoresSafeArea(.all, edges: .bottom) // Compose has own keyboard handler
       }
     }

swift file IOSApp

    import SwiftUI

    @main
    struct iOSApp: App {
    	    var body: some Scene {
		     WindowGroup {
			     ContentView()
		    }
	     }
     }

**Comment by xxfast** [comment]
Hi @adelsaramii Thanks for opening the issue.  You will need to have an `AppDelegate` defined like [this](https://github.com/xxfast/Decompose-Router/blob/main/app/ios/ios/AppDelegate.swift). Let me know if that helps

**Comment by xxfast** [comment]
Hi @adelsaramii apologies for the late reply. 

You'll need to add this file [`RouterContext.kt`](https://github.com/xxfast/Decompose-Router/blob/main/app/src/iosMain/kotlin/io/github/xxfast/decompose/router/app/utils/RouterContext.kt) in your shared module, under the iosMain source set 
```kotlin
fun defaultRouterContext(): RouterContext = defaultRouterContext()
val Lifecycle.registry get() = this as LifecycleRegistry
fun Lifecycle.destroy() = registry.destroy()
fun Lifecycle.resume() = registry.resume()
fun Lifecycle.stop() = registry.stop()
```
With that, `RouterContextKt` should no longer be an unresolved reference.

I think you raised a good issue here - it is not intuitive to set this up, and the documentation is unclear. I will use this issue to address this.

**Comment by voitenkodev** [comment]
Same problem for me 
``` 
import io.github.xxfast.decompose.router.defaultRouterContext
```
`Unresolved reference` for iOS target

**Comment by xxfast** [comment]
Hi @adelsaramii & @voitenkodev 

I've added [this](https://github.com/xxfast/Decompose-Router#ios) section in the readme, that covers how to setup decompose-router on ios - and added code-examples for kotlin app delegates, and both swift-ui app and swift uikit app delegate.

The `defaultRouterContext()` should now be available on the latest snapshot `0.5.1-SNAPSHOT`. Make sure to include the snapshot repository on your build if you wish to try out the snapshot builds - otherwise I will release stable 0.5.1 later today

Let me know how it goes

**Comment by xxfast** [comment]
Addressed in `0.5.1` 🎉
